### DISCUSSION TITLE
DDL change messages

### DISCUSSION THREAD

**Discussion by wolframm** [isoos/postgresql-dart#266] [Q&A] [answered]
Does the library have a `LogicalReplicationMessage` that will transmit DDL changes?

If not, is it feasible to catch `UnknownLogicalReplicationMessage` and decode that?

**Comment by isoos** [comment]
@wolframm: could you please elaborate more on what you are trying to do here?

/cc @osaxma

**Reply by osaxma** [reply]
I believe they are referring to DDL (Data Definition Language) commands such as `create`, `drop`, or `alter` that modifies the database structure.

**Comment by osaxma** [comment, marked as answer]
As far as I can tell, DDL commands are still not supported in logical replication according to the latest docs of version 16:

> Logical replication currently has the following restrictions or missing functionality. These might be addressed in future releases.
> - The database schema and DDL commands are not replicated.
>
> source: [Logical Replication Restrictions](https://www.postgresql.org/docs/16/logical-replication-restrictions.html)

**Reply by wolframm** [reply]
Of course. I confused logical with physical replication.
